File: WoodPeckerDemo/WoodPeckerDemo/Modules/Network/NetworkViewController.m
//
//  NetworkViewController.m
//  WoodPeckerDemo
//
//  Created by xiaogang zhang on 2018/1/17.
//  Copyright © 2018年 lifebetter. All rights reserved.
//

#import "NetworkViewController.h"

@interface NetworkViewController ()<NSURLConnectionDataDelegate, NSURLSessionDataDelegate>

@end

@implementation NetworkViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)startRequest:(id)sender {
    //url connection
    NSURL * requestURL = [NSURL URLWithString:@"https://api.github.com"];
    NSURLRequest * request = [NSURLRequest requestWithURL:requestURL];
    NSURLConnection * connection = [[NSURLConnection alloc] initWithRequest:request delegate:self];
    [connection start];
    
    //url session
    [self sessionRequest];
}

- (void)sessionRequest
{
    // NSURLSession
    NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
    config.timeoutIntervalForRequest = 10.0;
    NSURLSession *mySession = [NSURLSession sessionWithConfiguration:config delegate:self delegateQueue:[NSOperationQueue mainQueue]];
    NSString * url = @"https://api.github.com/emojis";
    NSURL * requestURL = [NSURL URLWithString:url];
    NSMutableURLRequest * uploadRequest = [NSMutableURLRequest requestWithURL:requestURL];
    uploadRequest.HTTPMethod = @"POST";
    NSString * formText = [NSString stringWithFormat:@"q=%.f",[[NSDate date] timeIntervalSince1970]];
    NSData * data = [formText dataUsingEncoding:NSUTF8StringEncoding];
    NSURLSessionTask * task = [mySession uploadTaskWithRequest:uploadRequest fromData:data completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
        
    }];
    [task resume];
}

- (nullable NSURLRequest *)connection:(NSURLConnection *)connection willSendRequest:(NSURLRequest *)request redirectResponse:(nullable NSURLResponse *)response
{
    return request;
}

- (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLResponse *)response
{
    
}

- (void)connection:(NSURLConnection *)connection didReceiveData:(NSData *)data
{
    
}

- (void)connection:(NSURLConnection *)connection
   didSendBodyData:(NSInteger)bytesWritten
 totalBytesWritten:(NSInteger)totalBytesWritten
totalBytesExpectedToWrite:(NSInteger)totalBytesExpectedToWrite
{
    
}

- (void)connectionDidFinishLoading:(NSURLConnection *)connection
{
    
}

- (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)dataTask didReceiveResponse:(NSURLResponse *)response completionHandler:(void (^)(NSURLSessionResponseDisposition))completionHandler
{
    completionHandler(NSURLSessionResponseAllow);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: WoodPeckerSwiftDemo/WoodPeckerSwiftDemo/Modules/Network/NetworkViewController.swift
//
//  NetworkViewController.swift
//  WoodPeckerSwiftDemo
//
//  Created by xiaogang zhang on 2018/1/21.
//  Copyright © 2018年 lifebetter. All rights reserved.
//

import UIKit

class NetworkViewController: UIViewController,NSURLConnectionDataDelegate,URLSessionDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    @IBAction func startRequest(_ sender: Any) {
        //url connection
        let requestURL = URL.init(string: "https://api.github.com")
        let request = URLRequest.init(url: requestURL!)
        let connection = NSURLConnection.init(request: request, delegate: self)
        connection?.start()
        //session request
        self.sessionRequest()
    }

    func sessionRequest(){
        let config  = URLSessionConfiguration.default
        config.timeoutIntervalForRequest = 10.0
        let mySession = URLSession.init(configuration: config, delegate: self, delegateQueue: OperationQueue.main)
        let url = "https://api.github.com/emojis"
        let requestURL = URL.init(string: url)
        let uploadRequest = NSMutableURLRequest.init(url: requestURL!)
        uploadRequest.httpMethod = "POST"
        let formText = "q=\(arc4random())"
        let data = formText.data(using: String.Encoding.utf8)
        let task = mySession.uploadTask(with: uploadRequest as URLRequest, from: data) { (data:Data?, response:URLResponse?, error:Error?) in
        }
        task.resume()
    }
    
    func connection(_ connection: NSURLConnection, willSend request: URLRequest, redirectResponse response: URLResponse?) -> URLRequest? {
        return request
    }
    
    func connection(_ connection: NSURLConnection, didReceive response: URLResponse) {
        
    }
    
    func connection(_ connection: NSURLConnection, didSendBodyData bytesWritten: Int, totalBytesWritten: Int, totalBytesExpectedToWrite: Int) {
        
    }

    func connection(_ connection: NSURLConnection, didReceive data: Data) {
        
    }
    
    func connectionDidFinishLoading(_ connection: NSURLConnection) {
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}






